### PR TITLE
Benchmark combinations of predefined poses

### PR DESCRIPTION
**Pull request by henningkayser**
Adds a benchmark executable that creates planner queries for all pair-wise combinations of named poses specified in the config.

### CODE REVIEW

**Comment by henningkayser** [comment]
I added an example and fixed some conflicts with changes in the previous PRs.
Currently the example depends on a new named target pose for the panda robot (https://github.com/ros-planning/panda_moveit_config).

File: moveit_ros/benchmarks/examples/demo_panda_predefined_poses.launch [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<launch>
```
Comment (davetcoleman) [inline moveit_ros/benchmarks/examples/demo_panda_predefined_poses.launch]: seems this should live in the tutorials... no?

File: moveit_ros/benchmarks/examples/demo_panda_predefined_poses.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# This is an example configuration that loads the "Kitchen" scene from the
+# local MoveIt! warehouse and benchmarks the "manipulator" group over all pairs
```
Comment (davetcoleman) [inline moveit_ros/benchmarks/examples/demo_panda_predefined_poses.yaml]: no !

File: moveit_ros/benchmarks/include/moveit/benchmarks/BenchmarkOptions.h [hunk 3]
```diff
@@ -68,6 +68,8 @@ class BenchmarkOptions
   const std::string& getGoalConstraintRegex() const;
   const std::string& getPathConstraintRegex() const;
   const std::string& getTrajectoryConstraintRegex() const;
+  const std::vector<std::string>& getPredefinedPoses() const;
```
Comment (davetcoleman) [inline moveit_ros/benchmarks/include/moveit/benchmarks/BenchmarkOptions.h]: even though no one else has, please add doxygen for your new functions

File: moveit_ros/benchmarks/examples/demo_panda_predefined_poses.launch (hunk 1 above)
Comment (henningkayser) [reply moveit_ros/benchmarks/examples/demo_panda_predefined_poses.launch]: Not sure about that. There is already a set of similar examples. It's also really convenient to be able to customize these from within the main package.

**Comment by rhaschke** [comment]
I rebased feature-benchmark-improvements onto latest master, to fix the latest Travis issue (moveit_msgs was used from melodic-devel branch, but moveit_visual_tools was using master features already). However, this requires to rebase this PR branch as well. @henningkayser, could you please do so?

File: moveit_ros/benchmarks/examples/demo_panda_predefined_poses.launch (hunk 1 above)
Comment (davetcoleman) [reply moveit_ros/benchmarks/examples/demo_panda_predefined_poses.launch]: ideally the tutorials would live in this repo (future work)

in other components we've been keeping stuff like this in the tutorials, but your call.

we could move all of these examples into the tutorials... this package predates the moveit_tutorials repo

File: moveit_ros/benchmarks/include/moveit/benchmarks/BenchmarkOptions.h [hunk 4]
```diff
@@ -48,31 +48,57 @@ namespace moveit_ros_benchmarks
...
 {
 public:
+  /** \brief Constructor */
   BenchmarkOptions();
+  /** \brief Constructor accepting a custom namespace for parameter lookup */
   BenchmarkOptions(const std::string& ros_namespace);
+  /** \brief Destructor */
   virtual ~BenchmarkOptions();
 
+  /** \brief Set the ROS namespace the node handle should use for parameter lookup */
```
Comment (davetcoleman) [inline moveit_ros/benchmarks/include/moveit/benchmarks/BenchmarkOptions.h]: awesome documentation!

**Pull request merged**